[IMP] general/payment_acquirers: add stripe connect documentation (#1447)
odoo/documentation: content/applications/finance/payment_acquirers/stripe.rst

Credentials tab
---------------

> The method to acquire your credentials depends on your hosting:

Reviewer: "of your hosting type" is enough IMO.

- On Odoo Online, all you need to do is follow the onboarding.
- On Odoo.sh and on-premise, extra steps are required.

Odoo Online
~~~~~~~~~~~

#. Go to the **eCommerce** or the **Sales** app and click on the *Activate Stripe* or the *Set

>    payments* button on the onboarding banner.

Reviewer: This *is* the normal way for Stripe Connect. There is another way but it has more friction. I'll add a tip.

Credentials tab
---------------

> The way you acquire your credentials depends on your hosting:

Reviewer 2: "The method to acquire"?

> - On Odoo Online, all you need to do is follow the onboarding.

Reviewer 2: I'd just say "Odoo Online, follow the onboarding" but that might be too harsh @jcs-odoo?

Reviewer: It hurts my feelings. I left "On Odoo Online, all you need to do is follow the onboarding." for now ;)

Reviewer 3: @LoredanaLrpz @AntoineVDV 

On Odoo Online, please follow the onboarding.

Otherwise.
If your hosting plan appears to be with our SaaS solution, namely Odoo Online, please consider following the user-friendly onboarding.

- On Odoo.sh and on-premise, extra steps are required.

Odoo Online
~~~~~~~~~~~

> #. Go to the **eCommerce** or **Sales** app and click on the *Activate Stripe* or *Set payments*
>    button on the onboarding banner.

Reviewer 2: ```suggestion
#. Go to the **eCommerce** or the **Sales** app and click on the *Activate Stripe* or
   the *Set payments* button on the onboarding banner.
```

> #. Fill in the requested information and submit the form once you are done.

Reviewer 2: ```suggestion
#. Fill in the requested information and submit the form.
```

#. Confirm your email address when Stripe sends you a confirmation email.
#. At the end of the process, you are redirected to Odoo. If you submitted all the requested
   information, you are all set and your payment acquirer is enabled.

.. tip::

>    If you prefer to use your own :ref:`API keys <stripe/api_keys>`, it is possible to do so by
>    :ref:`activating the Developer mode <developer-mode>` and :ref:`enabling Stripe manually
>    <payment_acquirers/add_new>`.

Reviewer 2: ```suggestion
   To use your own :ref:`API keys <stripe/api_keys>`, :ref:`activate the Developer mode
   <developer-mode>` and :ref:`enable Stripe manually <payment_acquirers/add_new>`.
```

On-premise or Odoo.sh
~~~~~~~~~~~~~~~~~~~~~

#. Go to the **eCommerce** or **Sales** app and click on the *Activate Stripe* or *Set payments*

>    button on the onboarding banner.

Reviewer 2: Same as 25-26

> #. Fill in the requested information and submit the form once you are done.

Reviewer 2: Same as 27

#. Confirm your email address when Stripe sends you a confirmation email.
#. At the end of the process, you are redirected to the payment acquirer **Stripe** on Odoo.
#. :ref:`Fill in your credentials <stripe/api_keys>` and :ref:`generate a webhook <stripe/webhook>`.
   Then, enable the payment acquirer.

.. important::
   If you are testing Stripe (in **test mode**), change the **State** to *Test Mode*. We recommend
   doing this on a test Odoo database rather than on your main database.

.. _stripe/api_keys:

Publishable and Secret keys
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Odoo needs your **API Credentials** to connect with your Stripe account, which comprises:

- :ref:`Publishable Key <stripe/api_keys>`: The key solely used to identify the account with Stripe.
- :ref:`Secret Key <stripe/api_keys>`: The key to sign the merchant account with Stripe.
- :ref:`Webhook Signing Secret <stripe/webhook>`: When you enable your webhook on your Stripe
  account, this signing secret must be set to authenticate the messages sent from Stripe to Odoo.

> 

Reviewer 2: I'd put all of this right after credentials and before line 17 (The way you acquire your credentials depends on your hosting:)

Reviewer: I'm not sure it's a good idea because you don't actually need these credentials if you're on Odoo Online.

Since this PR doesn't really touch these lines (git thinks it does but it only inserts new sections between the title and the lines), I'd prefer to leave them like this so they can be referenced from both "Odoo Online" and "Odoo.sh or on-premise" sections. Then, when content tabs become available in 15.0, we'll move the tip that references this section from the "Odoo Online" section inside the appropriate tab under "Publishable and Secret keys".

Reviewer 3: I agree it can stay afterward for now.
If they need an explanation on this, they can access this part of the page by clicking on the ref link.

Webhook Signing Secret
~~~~~~~~~~~~~~~~~~~~~~

To retrieve the webhook signing secret, create a webhook either automatically or manually.

Create the webhook automatically
********************************

Make sure your :ref:`Publishable and Secret keys <stripe/api_keys>` are filled in, then click on the
*Generate your Webhook* button.

Create the webhook manually
***************************

> Follow this `link to your webhooks <https://dashboard.stripe.com/webhooks>`_, or log into your

Reviewer 3: Better to have the link on a text about the link rather than on the word "link".

For example, Open you [link] (webhooks on Stripe)

Reviewer: I agree but (review tip) notice that this diff is not linked to the PR. It was simply moved and adapted to the change of context ;)

Reviewer 3: okééééééééé

Webhook Signing Secret
~~~~~~~~~~~~~~~~~~~~~~

> To retrieve the webhook signing secret, you first need to create a webhook. This can be done
> automatically or manually.

Reviewer 2: ```suggestion
To retrieve the webhook signing secret, create a webhook either automatically or manually.
```

Create the webhook automatically
********************************

Make sure your :ref:`Publishable and Secret keys <stripe/api_keys>` are filled in, then click on the
*Generate your Webhook* button.

Create the webhook manually
***************************

Follow this `link to your webhooks <https://dashboard.stripe.com/webhooks>`_, or log into your
Stripe dashboard and go to :menuselection:`Developers --> Webhooks`. Then, click on **Add endpoint**

> in your **Hosted endpoints**. A form opens, where you will need to add the following data:

Reviewer 2: ```suggestion
in your **Hosted endpoints** and insert the following data into the pop-up form:
```

Reviewer: 👌
